### ISSUE TITLE
Gilmour should log backtrace in per request errors only for code >= 500

### ISSUE THREAD

**Issue by adityagodbole** [gilmour-libs/gilmour#19]
The error logging below is excessive for error codes < 500

```
E, [2015-07-12T14:04:45.832564 #47673] ERROR -- 21d948b53c11e334db54bc3e58e4507d: 404 
"Object Not Found"
E, [2015-07-12T14:04:45.832773 #47673] ERROR -- 21d948b53c11e334db54bc3e58e4507d: 
["/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/net/http/response.rb:119:in `error!'",
 "/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/chef-12.3.0/lib/chef/http.rb:145:in `request'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/chef-12.3.0/lib/chef/rest.rb:119:in `get'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/chef-12.3.0/lib/chef/data_bag_item.rb:152:in
 `load'", "/Users/aditya/ds/chef-repo/.chef/plugins/knife/cluster_teardown.rb:44:in `run'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/chef-12.3.0/lib/chef/knife.rb:422:in `block in 
run_with_pretty_exceptions'", "/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/chef-
12.3.0/lib/chef/local_mode.rb:39:in `with_server_connectivity'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/chef-12.3.0/lib/chef/knife.rb:421:in `run_with_pretty_exceptions'", 
"/Users/aditya/ds/chef-repo/.chef/plugins/knife/common.rb:57:in `call_plugin'", "/Users/aditya/ds/chef-
repo/.chef/plugins/knife/common.rb:116:in `teardown_cluster'", "/Users/aditya/ds/chef-
repo/managers/cassandra_cluster.rb:353:in `do_teardown'", "/Users/aditya/ds/chef-
repo/managers/cassandra_cluster.rb:381:in `block in <class:CassandraSubscriber>'", 
"/Users/aditya/ds/chef-repo/.chef/plugins/knife/server.rb:81:in `call'", "/Users/aditya/ds/chef-
repo/.chef/plugins/knife/server.rb:81:in `block in ds_respond'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/responder.rb:183:in `instance_eval'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/responder.rb:183:in 
`block in _execute'", "/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/timeout.rb:91:in `block in timeout'",
 "/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/timeout.rb:35:in `block in catch'", 
"/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/timeout.rb:35:in `catch'", 
"/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/timeout.rb:35:in `catch'", 
"/Users/aditya/.rbenv/versions/2.1.5/lib/ruby/2.1.0/timeout.rb:106:in `timeout'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/responder.rb:182:in `_execute'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/responder.rb:102:in 
`block in execute'", "/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/gilmour-
0.3.0/lib/gilmour/responder.rb:96:in `fork'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/responder.rb:96:in `execute'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/gilmour-
0.3.0/lib/gilmour/backends/backend.rb:110:in `_execute_handler'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/gilmour-0.3.0/lib/gilmour/backends/backend.rb:98:in `block in 
execute_handler'", "/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/gilmour-
0.3.0/lib/gilmour/backends/redis.rb:142:in `block (2 levels) in acquire_ex_lock'", "/Users/aditya/ds/chef-
repo/vendor/bundle/ruby/2.1.0/gems/eventmachine-1.0.7/lib/eventmachine.rb:1054:in `call'", 
"/Users/aditya/ds/chef-repo/vendor/bundle/ruby/2.1.0/gems/eventmachine-
1.0.7/lib/eventmachine.rb:1054:in `block in spawn_threadpool'"]
```

**Comment by adityagodbole** [comment]
This has been implemented in commit a9c900b0

**Comment by meson10** [comment]
Oh I never realised. Thanks.
